include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -736,13 +954,16 @@ oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy,
 reduce_by_segment(_ExecutionPolicy&& __exec, _Range1&& __keys, _Range2&& __values, _Range3&& __out_keys,
                   _Range4&& __out_values)
 {
-    return reduce_by_segment(::std::forward<_ExecutionPolicy>(__exec), ::std::forward<_Range1>(__keys),
-                             ::std::forward<_Range2>(__values), ::std::forward<_Range3>(__out_keys),
-                             ::std::forward<_Range4>(__out_values), oneapi::dpl::__internal::__pstl_equal());
+    return reduce_by_segment(std::forward<_ExecutionPolicy>(__exec), std::forward<_Range1>(__keys),
+                             std::forward<_Range2>(__values), std::forward<_Range3>(__out_keys),
+                             std::forward<_Range4>(__out_values), oneapi::dpl::__internal::__pstl_equal());
 }
 
 } // namespace ranges
 } // namespace experimental
+
+#endif //_ONEDPL_HETERO_BACKEND
```

Comment:
Why do we have that in `glue_`? Is it because we are mixing experimental and non-experimental API in one file?

Reply:
It because I did not see any sense to introduce a new file.